lib/shelf_proxy.dart

```diff
@@ -30,7 +30,7 @@ Handler proxyHandler(url, {http.Client client, String proxyName}) {
   } else {
     throw ArgumentError.value(url, 'url', 'url must be a String or Uri.');
   }
-  client ??= http.Client();
+  final theClient = client ?? http.Client();
```

Comment:
I'm a little surprised we need this. Was the particular reason to not reassign to the parameter?

Reply:
This is the crazy function wrapper promotion silly. Figured this is better than putting a `client!` in the lambda

Reply:
What do you think of the name `nonNullClient` so the intent is more clear?

Reply:
Sure. But now you get a bunch of lints, too!